core/src/components/datetime/utils/data.ts

```diff
@@ -166,7 +167,7 @@ export const generateTime = (
     processedMinutes = processedMinutes.filter((minute) => minuteValues.includes(minute));
   }
 
-  if (minParts) {
+  if (minParts && currentValue !== undefined) {
```

Comment:
I don't think this function should be tightly coupled with the state of the `value` property. Maybe we could just not pass in the max/min parts when `value === undefined`? That would result in all possible times being generated.